ocp_resources/storage_class.py

```diff
@@ -45,3 +45,57 @@ class Annotations:
         IS_DEFAULT_CLASS = (
             f"{Resource.ApiGroup.STORAGECLASS_KUBERNETES_IO}/is-default-class"
         )
+
+    class ReclaimPolicy:
+        DELETE = "Delete"
+        RETAIN = "Retain"
+
+    def __init__(
+        self,
+        name=None,
+        client=None,
+        teardown=True,
+        provisioner=None,
+        reclaim_policy=None,
+        volume_binding_mode=None,
+        allow_volume_expansion=None,
+        parameters=None,
+        **kwargs,
+    ):
+        """
+        StorageClass object
+
+        Args:
+            name (str): StorageClass name.
+            client (DynamicClient): DynamicClient to use.
+            teardown (bool): Indicates if this resource would need to be deleted.
+            provisioner (str): The provisioner of the storage class
+            reclaim_policy (str): Can be either "Delete" or "Retain"
+            volume_binding_mode (str): When volume binding and dynamic provisioning should occur
+            allow_volume_expansion (bool): True for allowing the volume expansion
+            parameters (list): Describe volumes belonging to the storage class.
+        """
+        super().__init__(
+            name=name,
+            client=client,
+            teardown=teardown,
+            **kwargs,
+        )
+        self.provisioner = provisioner
+        self.reclaim_policy = reclaim_policy
+        self.volume_binding_mode = volume_binding_mode
+        self.allow_volume_expansion = allow_volume_expansion
+        self.parameters = parameters
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res.update({"provisioner": self.provisioner})
```

Comment:
is `self.provisioner` mandatory? 
if so please raise if it isn't passed

Reply:
ack